### ISSUE TITLE
yarn build fails

### ISSUE THREAD

**Issue by dietmarw** [xogeny/ModelicaBook#397]
I tried to generate the site locally using `make site` but get the following error:

```
docker run -v `pwd`:/opt/MBE/ModelicaBook -w /opt/MBE/ModelicaBook/generator -i -t mtiller/book-builder yarn build
yarn run v1.3.2
$ DEBUG='mbe:*' react-static build
{ Error: Cannot find module './json/_static/sponsors/sponsors.json'
[...]
```
looking for `sponsors.json` I find it in the following locations relative from the ModelicaBook root:

```
ModelicaBook/apps/public/_static/sponsors/sponsors.json
ModelicaBook/sponsors/sponsors.json
ModelicaBook/text/build/dirhtml/_static/sponsors/sponsors.json
```
none of which resides under a `json` dir where yarn seems to look for it. So I'm puzzled why it works for you @mtiller.

**Comment by xogeny** [comment]
Definitely works for me.  I'll investigate why I don't get that error...

**Comment by xogeny** [comment]
OK, so I know why it works for me.  Question is why it doesn't work for you.  You say that the only place you see `sponsors.json` inside `ModelicaBook` is in those three directories.  But I have:

```
./text/build/json/_static/sponsors/sponsors.json
```

...as well.  This seems to indicate you didn't run the `json` target for Sphinx.  The issue, I suspect, is that building the site normally is done through CircleCI.  In that case, the `circle.yml` file is used and *that* runs `make json`.  You probably were just doing `make all` which (until now) skipped that step.  But I fixed that in bebf2b2.

So assuming I'm correct, all you need to do is `make json` or (now) `make all` before `make site`.

Does that help?

**Comment by dietmarw** [comment]
Yes that helped. But your fix is not sufficient in a way. If `make site` depends on the `json` rule that it should be put in as dep there and not just in the `all` rule. So running any of the rules should succeed without having to manually figuring out the dependencies by trial and error. Do you agree?
